waku/v2/protocol/peer_exchange/waku_peer_exchange_test.go

```diff
@@ -185,3 +213,129 @@ func TestRetrieveFilteredPeerExchangePeers(t *testing.T) {
 	px1.Stop()
 	px3.Stop()
 }
+
+func TestPeerExchangeOptions(t *testing.T) {
+
+	// Prepare host1
+	host1, _, prvKey1 := tests.CreateHost(t)
+	udpPort1, err := tests.FindFreePort(t, "127.0.0.1", 3)
+	require.NoError(t, err)
+	ip1, _ := tests.ExtractIP(host1.Addrs()[0])
+	l1, err := tests.NewLocalnode(prvKey1, ip1, udpPort1, wenr.NewWakuEnrBitfield(false, false, false, true), nil, utils.Logger())
+	require.NoError(t, err)
+	discv5PeerConn1 := discv5.NewTestPeerDiscoverer()
+	d1, err := discv5.NewDiscoveryV5(prvKey1, l1, discv5PeerConn1, prometheus.DefaultRegisterer, utils.Logger(), discv5.WithUDPPort(uint(udpPort1)))
+	require.NoError(t, err)
+	d1.SetHost(host1)
+
+	// Mount peer exchange
+	pxPeerConn1 := discv5.NewTestPeerDiscoverer()
+	px1, err := NewWakuPeerExchange(d1, pxPeerConn1, nil, prometheus.DefaultRegisterer, utils.Logger())
+	require.NoError(t, err)
+	px1.SetHost(host1)
+
+	// Test WithPeerAddr()
+	params := new(PeerExchangeParameters)
+	params.host = px1.h
+	params.log = px1.log
+	params.pm = px1.pm
+
+	optList := DefaultOptions(px1.h)
+	optList = append(optList, WithPeerAddr(host1.Addrs()[0]))
+	for _, opt := range optList {
+		err := opt(params)
+		require.NoError(t, err)
+	}
+
+	require.Equal(t, host1.Addrs()[0], params.peerAddr)
+
+	// Test WithFastestPeerSelection()
+	optList = DefaultOptions(px1.h)
+	optList = append(optList, WithFastestPeerSelection(host1.ID()))
+	for _, opt := range optList {
+		err := opt(params)
+		require.NoError(t, err)
+	}
+
+	require.Equal(t, peermanager.LowestRTT, params.peerSelectionType)
+	require.Equal(t, host1.ID(), params.preferredPeers[0])
+
+}
+
+func TestRetrieveProvidePeerExchangeWithPMAndPeerAddr(t *testing.T) {
+	log := utils.Logger()
+
+	// H1 + H2 with discovery on
+	host1, d1 := createHostWithDiscv5(t)
+	host2, d2 := createHostWithDiscv5(t, d1.Node())
+
+	// H3
+	ps3, err := pstoremem.NewPeerstore()
+	require.NoError(t, err)
+	wakuPeerStore3 := wps.NewWakuPeerstore(ps3)
+	host3, _, _ := tests.CreateHost(t, libp2p.Peerstore(wakuPeerStore3))
+
+	defer d1.Stop()
+	defer d2.Stop()
+	defer host1.Close()
+	defer host2.Close()
+	defer host3.Close()
+
+	err = d1.Start(context.Background())
+	require.NoError(t, err)
+
+	err = d2.Start(context.Background())
+	require.NoError(t, err)
+
+	// Prepare peer manager for host3
+	pm3 := peermanager.NewPeerManager(10, 20, log)
+	pm3.SetHost(host3)
+	pxPeerConn3, err := peermanager.NewPeerConnectionStrategy(pm3, 30*time.Second, utils.Logger())
+	require.NoError(t, err)
+	pm3.SetPeerConnector(pxPeerConn3)
```

Comment:
This line is not required as this is done automatically inside the `NewPeerConnectionStrategy` method.

Reply:
Fixed at https://github.com/waku-org/go-waku/pull/1046/commits/f0cfb6a4880bb7048e18018d8b507ae24c75ad4f